### PR TITLE
feat: client intialized with ticket and CSRF prevention Token

### PR DESCRIPTION
**Pull request by camaeel**
Client intialized with ticket and CSRF prevention Token.

Completes #370

### CODE REVIEW

File: proxmox/client.go [hunk 1]
```diff
@@ -131,6 +131,10 @@ func (c *Client) SetAPIToken(userID, token string) {
 	c.session.SetAPIToken(userID, token)
 }
 
+func (c *Client) SetTicket(ticket, csrfPreventionToken string) {
```
Comment (Tinyblargon) [inline proxmox/client.go:142]: Could you add a comment to this?
Just a notice that it is different from the normal login/api authentication methods and what it is used for.

**Comment by camaeel** [comment]
@Tinyblargon any chance for a review and merge for this tiny PR?

File: proxmox/session.go [hunk 2]
```diff
@@ -167,6 +167,11 @@ func (s *Session) SetAPIToken(userID, token string) {
 	s.AuthToken = auth
 }
 
+func (s *Session) SetTicket(ticket, csrfPreventionToken string) {
```
Comment (Tinyblargon) [inline proxmox/session.go]: Can you make this private so `setTicket`? Nothing in `Session` should be exposed.

**Review by Tinyblargon — CHANGES_REQUESTED** [review]
Could you make the 2 minor changes?
Everything else looks good.

Sorry, thought I had finished the code review already.

File: proxmox/client.go (hunk 1 above)
Comment (camaeel) [reply proxmox/client.go:142]: Addressed

File: proxmox/session.go (hunk 2 above)
Comment (camaeel) [reply proxmox/session.go]: func (Session)setTicket made private

**Comment by camaeel** [comment]
Thanx @Tinyblargon

**Pull request merged**